src/optimization/hooke_jeeves_method.cpp

```diff
@@ -0,0 +1,157 @@
+#include <k52/optimization/hooke_jeeves_method.h>
+
+#ifdef BUILD_WITH_MPI
+
+#include <boost/mpi.hpp>
+#include <k52/parallel/mpi/constants.h>
+
+#endif
+
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+#include <stdexcept>
+
+#include <k52/optimization/params/i_continuous_parameters.h>
+
+using ::std::vector;
+
+namespace k52
+{
+namespace optimization
+{
+
+HookeJeevesMethod::HookeJeevesMethod(
+	double acceleration,
+	double init_step,
+	size_t max_iteration_number,
+	double precision)
+	: acceleration_(acceleration)
+	, init_step_(init_step)
+	, max_iteration_number_(max_iteration_number)
+	, precision_(precision)
+{
+}
+
+HookeJeevesMethod* HookeJeevesMethod::Clone() const
+{
+	return new HookeJeevesMethod(
+		acceleration_,
+		init_step_,
+		max_iteration_number_,
+		precision_);
+}
+
+std::string HookeJeevesMethod::get_name() const
+{
+	return "Hooke-Jeeves Method";
+}
+
+#ifdef BUILD_WITH_MPI
+void HookeJeevesMethod::Send(boost::mpi::communicator* communicator, int target) const
+{
+	communicator->send(target, k52::parallel::mpi::constants::kCommonTag, acceleration_);
+	communicator->send(target, k52::parallel::mpi::constants::kCommonTag, precision_);
+	communicator->send(target, k52::parallel::mpi::constants::kCommonTag, max_iteration_number_);
+	communicator->send(target, k52::parallel::mpi::constants::kCommonTag, init_step_);
+}
+
+void HookeJeevesMethod::Receive(boost::mpi::communicator* communicator, int source)
+{
+	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, acceleration_);
+	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, precision_);
+	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, max_iteration_number_);
+	communicator->recv(source, k52::parallel::mpi::constants::kCommonTag, init_step_);
+}
+#endif
+
+vector<double> HookeJeevesMethod::FindOptimalParameters(const vector<double>& initial_parameters)
+{
+	vector<double> arguments(initial_parameters);
+	vector<double> arguments_after_search(initial_parameters.size());
+	vector<double> steps_array = vector<double>(arguments.size(), init_step_);
+	size_t iteration = 0;
+
+	do
+	{
+		arguments_after_search = CoordinatewiseSearch(arguments, steps_array);
+		if (arguments == arguments_after_search)
```

Comment:
Invalid comparison of floating point values. Potentially an endless loop.
See k52.common.FloatingPoint. You need to add corresponding method for vector and reuse existing AreEqual there, also add corresponding unit tests.